# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1280501304 14400
#      Fri Jul 30 10:48:24 2010 -0400
# Node ID 6abf966379dec3ad4180810e3adacd17d56cbbf5
# Parent  4bbd6f75f173952667d2f6c15feaab014dffdbd0
pt-eval.cc: process breakpoints for if and switch commands

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,14 @@
+2010-07-30  John W. Eaton  <jwe@octave.org>
+
+	* pt-eval.cc (tree_evaluator::visit_if_command,
+	tree_evaluator::visit_switch_command): Also process breakpoints
+	for the command (bug #30490).
+
 2010-07-29  Ben Abbott <bpabbott@mac.com>
 
 	* gl-render.cc: Transpose markers '^' and 'v'.
 
 2010-07-29  John W. Eaton  <jwe@octave.org>
 
 	* pt-bp.cc (tree_breakpoint::visit_if_command): Also allow
 	breakpoint to be set on command, same as for switch, while, or
diff --git a/src/pt-eval.cc b/src/pt-eval.cc
--- a/src/pt-eval.cc
+++ b/src/pt-eval.cc
@@ -537,16 +537,19 @@ void
 tree_evaluator::visit_if_clause (tree_if_clause&)
 {
   panic_impossible ();
 }
 
 void
 tree_evaluator::visit_if_command (tree_if_command& cmd)
 {
+  if (debug_mode)
+    do_breakpoint (cmd.is_breakpoint ());
+
   tree_if_command_list *lst = cmd.cmd_list ();
 
   if (lst)
     lst->accept (*this);
 }
 
 void
 tree_evaluator::visit_if_command_list (tree_if_command_list& lst)
@@ -803,16 +806,19 @@ void
 tree_evaluator::visit_switch_case_list (tree_switch_case_list&)
 {
   panic_impossible ();
 }
 
 void
 tree_evaluator::visit_switch_command (tree_switch_command& cmd)
 {
+  if (debug_mode)
+    do_breakpoint (cmd.is_breakpoint ());
+
   tree_expression *expr = cmd.switch_value ();
 
   if (expr)
     {
       octave_value val = expr->rvalue1 ();
 
       tree_switch_case_list *lst = cmd.case_list ();
 
